include/revng-c/mlir/Dialect/Clift/IR/CliftOps.td

```diff
@@ -141,73 +141,125 @@ def Clift_UndefOp: Clift_Op<"undef"> {
 
 //===----------------------------- Statements -----------------------------===//
 
-def Clift_LoopOp : Clift_Op<"loop", [Terminator]> {
-  let regions = (region AnyRegion:$body);
-  let successors = (successor
-    VariadicSuccessor<AnySuccessor>:$LoopDestinations);
+def Clift_AssignLabelOp : Clift_Op<"assign_label",
+                                   [Clift_StatementOpInterface]> {
+  let arguments = (ins Clift_LabelType:$label);
 
   let assemblyFormat = [{
-    $body $LoopDestinations attr-dict
+    $label type($label) attr-dict
   }];
-
-  let hasRegionVerifier = 1;
 }
 
-def Clift_ContinueOp : Clift_Op<"continue", [HasParent<"LoopOp">, Terminator]> {}
-
-def Clift_LabelResource : Resource<"LabelResource">;
-
-def Clift_AssignLabelOp : Clift_Op<"assign_label"> {
-  let arguments = (ins Arg<Clift_LabelType, "", [MemFree]>:$label);
+def Clift_GoToOp : Clift_Op<"goto",
+                            [Clift_StatementOpInterface,
+                             Clift_NoFallthrough]> {
+  let arguments = (ins Clift_LabelType:$label);
 
   let assemblyFormat = [{
     $label type($label) attr-dict
   }];
+}
 
-  let hasCanonicalizeMethod = 1;
+def Clift_IfOp : Clift_Op<"if",
+                          [Clift_StatementOpInterface]> {
+  let regions = (region Clift_ExpressionRegion:$condition,
+                        Clift_StatementRegion:$then,
+                        Clift_StatementRegion:$else);
+
+  let assemblyFormat = [{
+    $condition $then (`else` $else^)? attr-dict
+  }];
 }
 
-def Clift_GoToOp : Clift_Op<"goto", [Terminator]> {
-  let arguments = (ins Arg<Clift_LabelType, "", [MemRead]>:$label);
+def Clift_LocalVariableOp : Clift_Op<"local",
+                                     [Clift_StatementOpInterface]> {
+  let arguments = (ins StrAttr:$sym_name,
+                       TypeAttrOf<Clift_ValueType>:$type);
+
+  let regions = (region Clift_ExpressionRegion:$initializer);
 
   let assemblyFormat = [{
-    $label type($label) attr-dict
+    $type (`=` $initializer^)? attr-dict
   }];
 }
 
-def Clift_MakeLabelOp : Clift_Op<"make_label", [Clift_OneUseOfType<"AssignLabelOp">] > {
-  let results = (outs Res<Clift_LabelType, "", [MemAlloc, MemWrite] >:$result);
+def Clift_MakeLabelOp : Clift_Op<"make_label",
```

Comment:
Can you keep this close to `goto` and `assign_label`? They belong together.

Reply:
I just ordered them alphabetically, but sure I can do that.